xlive/Blam/Engine/interface/user_interface_text.cpp

```diff
@@ -74,6 +78,122 @@ bool __cdecl user_interface_parse_string(wchar_t* string, size_t max_length, cha
 	return INVOKE(0x22F712, 0x0, user_interface_parse_string, string, max_length, a3);
 }
 
+void user_interface_get_key_character(e_input_key_code key_code, c_static_wchar_string<512>* string)
+{
+	c_maximum_interface_text src;
+
+	ascii_key* key = &ascii_to_key_table[input_windows_get_remapped_key(key_code)];
+	switch (key_code)
+	{
+	case _key_oem_8:
+		if (get_current_language() == _language_french)
+		{
+			user_interface_global_string_get(_string_id_key_exclamation, &src);
+			string->set(src.get_string());
+		}
+		break;
+	case _key_tilde:
+		if (get_current_language() == _language_italian)
+		{
+			user_interface_global_string_get(_string_id_key_grave_o, &src);
+			string->set(src.get_string());
+		}
+		if (get_current_language() == _language_french)
+		{
+			user_interface_global_string_get(_string_id_key_grave_u, &src);
+			string->set(src.get_string());
+		}
+		break;
+	case _key_left_square_bracket:
+		if (get_current_language() == _language_german)
+		{
+			user_interface_global_string_get(_string_id_key_beta, &src);
+			string->set(src.get_string());
+		}
+		break;
+	case _key_right_square_bracket:
+		if (get_current_language() == _language_french)
+		{
+			user_interface_global_string_get(_string_id_key_acute, &src);
+			string->set(src.get_string());
+		}
+		break;
+	case _key_backslash:
+		if (get_current_language() == _language_japanese)
+		{
+			user_interface_global_string_get(_string_id_key_yen, &src);
+			string->set(src.get_string());
+		}
+		if (get_current_language() == _language_french)
+		{
+			user_interface_global_string_get(_string_id_key_asterisk, &src);
+			string->set(src.get_string());
+		}
+		break;
+	case _key_forwardslash:
+		if (get_current_language() == _language_italian)
+		{
+			user_interface_global_string_get(_string_id_key_grave_u, &src);
+			string->set(src.get_string());
+		}
+		if (get_current_language() == _language_french)
+		{
+			user_interface_global_string_get(_string_id_key_colon, &src);
+			string->set(src.get_string());
+		}
+		break;
+	case _key_colon:
+		if (get_current_language() == _language_french)
+		{
+			user_interface_global_string_get(_string_id_key_dollar, &src);
+			string->set(src.get_string());
+		}
+		if (get_current_language() == _language_italian)
```

Comment:
else if, they cannot be the same at the same time, and do it for the other instances.